document_page/models/document_page.py

```diff
@@ -0,0 +1,176 @@
+        "page_id",
+        "History",
+        order="create_date DESC",
+        readonly=True,
+    )
+    menu_id = fields.Many2one("ir.ui.menu", "Menu", readonly=True)
+    content_date = fields.Datetime(
+        "Last Contribution Date",
+        related="history_head.create_date",
+        store=True,
+        index=True,
+        readonly=True,
+    )
+    content_uid = fields.Many2one(
+        "res.users",
+        "Last Contributor",
+        related="history_head.create_uid",
+        store=True,
+        index=True,
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        help="If set, page is accessible only from this company",
+        index=True,
+        ondelete="cascade",
+    )
+    backend_url = fields.Char(
+        string="Backend URL",
+        help="Use it to link resources univocally",
+        compute="_compute_backend_url",
+    )
+
+    @api.depends("menu_id", "parent_id.menu_id")
+    def _compute_backend_url(self):
+        tmpl = "/web#id={}&model=document.page&view_type=form"
+        for rec in self:
+            url = tmpl.format(rec.id)
+            # retrieve action
+            action = None
+            parent = rec
+            while not action and parent:
+                action = parent.menu_id.action
+                parent = parent.parent_id
+            if action:
+                url += "&action={}".format(action.id)
+            rec.backend_url = url
+
+    @api.constrains("parent_id")
+    def _check_parent_id(self):
+        if not self._check_recursion():
+            raise ValidationError(_("You cannot create recursive categories."))
+
+    def _get_page_index(self, link=True):
+        """Return the index of a document."""
+        self.ensure_one()
+        index = []
+        for subpage in self.child_ids:
+            index += ["<li>" + subpage._get_page_index() + "</li>"]
+        r = ""
+        if link:
+            r = '<a href="{}">{}</a>'.format(self.backend_url, self.name)
+        if index:
+            r += "<ul>" + "".join(index) + "</ul>"
+        return r
+
+    @api.depends("history_head")
+    def _compute_content(self):
+        for rec in self:
+            if rec.type == "category":
+                rec.content = rec._get_page_index(link=False)
+            else:
+                if rec.history_head:
+                    rec.content = rec.history_head.content
+                else:
+                    # html widget's default, so it doesn't trigger ghost save
+                    rec.content = "<p><br></p>"
+
+    def _inverse_content(self):
+        for rec in self:
+            if rec.type == "content" and rec.content != rec.history_head.content:
+                rec._create_history(
+                    {
+                        "name": rec.draft_name,
+                        "summary": rec.draft_summary,
+                        "content": rec.content,
+                    }
+                )
```

Comment:
@Tardo I made some changes here but I didn't use the filtered function because anyway it would be necessary to iterate through each element after filtering to obtain the values ​​to create the corresponding document.page.history.
Anyway check it out.